nexus/sdk-go.md

```diff
@@ -0,0 +1,352 @@
+func NewWorkflowRunOperationWithOptions[I, O any](opts WorkflowRunOperationOptions[I, O]) nexus.Operation[I, O]
+
+type WorkflowHandle[T any] interface {
+	ID() string
+	RunID() string
+}
+
+// ExecuteWorkflow starts a workflow run, linking the execution chain to a Nexus operation (subsequent runs started from
+// continue-as-new and retries).
+// Automatically propagates the callback and request ID from the nexus options to the workflow.
+func ExecuteWorkflow[I, O any, WF func(internal.Context, I) (O, error)](
+	ctx context.Context,
+	nexusOptions nexus.StartOperationOptions,
+	startWorkflowOptions client.StartWorkflowOptions,
+	workflow WF,
+	arg I,
+) (WorkflowHandle[O], error)
+
+// ExecuteUntypedWorkflow starts a workflow with by function reference or string name, linking the execution chain to a
+// Nexus operation.
+// Useful for invoking workflows that don't follow the single argument - single return type signature.
+// See [ExecuteWorkflow] for more information.
+func ExecuteUntypedWorkflow[R any](
+	ctx context.Context,
+	nexusOptions nexus.StartOperationOptions,
+	startWorkflowOptions client.StartWorkflowOptions,
+	workflow any,
+	args ...any,
+) (WorkflowHandle[R], error)
+```
+
+**Usage**:
+
+```go
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporalnexus"
+)
+
+func MyHandlerWorkflow(workflow.Context, MyInput) (MyOutput, error)
+func MyHandlerWorkflowWithAlternativeInput(workflow.Context, MyWorkflowInput) (MyOutput, error)
+
+// Alternative 1 - shortest form, for workflows that have input and outputs that map 1:1 with the operation's I/O.
+opStartTransactionAlt1 := temporalnexus.NewWorkflowRunOperation(
+	"start-transaction",
+	MyHandlerWorkflow,
+	func(ctx context.Context, input MyInput, opts nexus.StartOperationOptions) (client.StartWorkflowOptions, error) {
+		return client.StartWorkflowOptions{
+			ID: input.ID,
+		}, nil
+	})
+
+// Alternative 2 - same as above but using the "WithOptions" method.
+opStartTransactionAlt2 := temporalnexus.NewWorkflowRunOperationWithOptions(
+	temporalnexus.WorkflowRunOperationOptions[MyInput, MyOutput]{
+		Name: "start-transaction",
+		Workflow: MyHandlerWorkflow,
+		OptionsGetter: func(ctx context.Context, input MyInput, opts nexus.StartOperationOptions) (client.StartWorkflowOptions, error) {
+			return client.StartWorkflowOptions{
+				ID: input.ID,
+			}, nil
+		},
+	})
+
+// Alternative 3 - start a workflow with alternative inputs.
+opStartTransactionAlt3 := temporalnexus.NewWorkflowRunOperationWithOptions(
+	temporalnexus.WorkflowRunOperationOptions[MyInput, MyOutput]{
+		Name: "start-transaction",
+		Handler: func(ctx context.Context, input MyInput, opts nexus.StartOperationOptions) (temporalnexus.WorkflowHandle[MyOutput], error) {
+			// Workflows started with this API must take a single input and return single output.
+			return temporalnexus.ExecuteWorkflow(ctx, client.StartWorkflowOptions{
```

Comment:
This doesn't match the signature of `ExecuteWorkflow` put above. Where's the `nexus.StartOperationOptions` parameter?

Reply:
Oh I added that arg last minute and forgot to update the example.
Thanks for pointing it out.
I was debating taking the options from the context instead of forcing users to pass those in but I think the signature with the operation options is a better approach.

Reply:
Fixed.